cuda_core/cuda/core/experimental/_memoryview.pyx

```diff
@@ -92,9 +102,52 @@ cdef class StridedMemoryView:
             else:
                 view_as_cai(obj, stream_ptr, self)
         else:
-            # default construct
             pass
 
+    @property
+    def shape(self) -> tuple[int]:
+        if self._shape is None and self.exporting_obj is not None:
+            if self.dl_tensor != NULL:
+                self._shape = cuda_utils.carray_int64_t_to_tuple(
+                    self.dl_tensor.shape, 
+                    self.dl_tensor.ndim
+                )
+            else:
+                self._shape = self.metadata["shape"]
+        else:
+            self._shape = ()
+        return self._shape
+
+    @property
+    def strides(self) -> Optional[tuple[int]]:
+        cdef int itemsize
+        if self._strides is None and self.exporting_obj is not None:
```

Comment:
I see a minor potential concern here. Unlike `_shape` where we can use `None` to indicate it's uninitialized, `_strides` cannot do this because `None` is a valid value (indicating C-contiguity) which was meant to be a fast path. Now with this check, we turn the fast path into a slow path because we keep re-running the checks (and resetting `_strides` to `None`) every time this property is requested. A dumb fix would be to add another boolean member to indicate if `_strides` is initialized. Thoughts?